core/src/com/unciv/logic/GameStarter.kt

```diff
@@ -580,21 +581,36 @@ object GameStarter {
     ): HashMap<Civilization, Tile>? {
         val startingLocations = HashMap<Civilization, Tile>()
         for (civ in civsOrderedByAvailableLocations) {
+            
+            val startingLocation = getCivStartingLocation(civ, tileMap, freeTiles, startScores)
+            startingLocations[civ] = startingLocation
+            
             val distanceToNext = minimumDistanceBetweenStartingLocations /
                 (if (civ.isCityState()) 2 else 1) // We allow city states to squeeze in tighter
-            val presetStartingLocation = tileMap.startingLocationsByNation[civ.civName]?.randomOrNull()
-            val startingLocation = if (presetStartingLocation != null) presetStartingLocation
-            else {
-                if (freeTiles.isEmpty()) break // we failed to get all the starting tiles with this minimum distance
-                getOneStartingLocation(civ, tileMap, freeTiles, startScores)
-            }
-            startingLocations[civ] = startingLocation
             freeTiles.removeAll(tileMap.getTilesInDistance(startingLocation.position, distanceToNext)
                 .toSet())
         }
         return if (startingLocations.size < civsOrderedByAvailableLocations.size) null else startingLocations
     }
 
+    private fun getCivStartingLocation(
+        civ: Civilization,
+        tileMap: TileMap,
+        freeTiles: MutableList<Tile>,
+        startScores: HashMap<Tile, Float>,
+    ): Tile {
+        var startingLocation = tileMap.startingLocationsByNation[civ.civName]?.randomOrNull()
+        if (startingLocation == null) {
+            startingLocation = tileMap.startingLocationsByNation[Constants.spectator]?.randomOrNull()
+            if (startingLocation != null) {
+                tileMap.startingLocationsByNation[Constants.spectator]?.remove(startingLocation)
+            }
+        }
+        if (startingLocation == null && freeTiles.isNotEmpty())
+            startingLocation = getOneStartingLocation(civ, tileMap, freeTiles, startScores)
+        // If startingLocation is null we failed to get all the starting tiles with this minimum distance
```

Comment:
Won't this throw a type cast error? If failure is an option, and it seems it is, this should return null, so our type result should be 'Tile?'